src/Glpi/Migration/AbstractPluginMigration.php

```diff
@@ -91,17 +122,22 @@ final public function execute(bool $simulate = false): PluginMigrationResult
     {
         $this->result = new PluginMigrationResult();
 
+        // Use a transaction unless we are already inside one.
+        $use_transaction = !$this->db->inTransaction();
```

Comment:
I do not like this change. We should not complexify the code this way, just to test that the expected configuration entry has been updated.

You can probably mock `$this->db` using `onlyMethods(['beginTransaction', 'commit', 'rollBack'])`.

Reply:
Tbh, the real issue is in our database implementation.

Nested transactions is a common issue which is fixed by SQL save points. 
`$DB->beginTransaction()` should not throw an error in case of a transaction but it should create a savepoint instead.

This is also what doctrine does: https://www.doctrine-project.org/projects/doctrine-dbal/en/4.3/reference/transactions.html#transaction-nesting

Reply:
I'll do some digging to see if it is easy to implement, shouldn't be too hard. Looking into it.